packages/useValidators/README.md

```diff
@@ -0,0 +1,44 @@
+# useValidators
+
+This is a hook that allows the definition of validators that can be used to run against a provided value. 
+
+# Installation
+
+### npm
+```bash
+npm install @es-hooks/useValidators
+```
+
+### yarn
+```bash
+yarn add @es-hooks/useValidators
+```
+
+# Usage
+
+```jsx
+import React, { useContext } from 'react'
+import useValidators from '@es-hooks/useValidators'
+
+function MyComponent(props) {
+  function isOverThreeCharacters(value) {
+    return value.length > 3 ? true : false
+  }
+
+  const [ValidationContext, invokeValidators] = useValidators([isOverThreeCharacters])
+  const results = useContext(ValidationContext)
+
+  const isValid = results.every(result => result !== false)
+
+  return (
+    <>
+      <label htmlFor="firstName">First name</label>
+      <input type="text" id="firstName" onChange={e => invokeValidators(e.target.value)} />
+
+      <h1>{isValid ? Valid! : Invalid!}</h1>
```

Comment:
Is it worth making an example of advanced usage in a form with multiple inputs? Let's take for example we add a `lastName` input to the above. I can see a user thinking it would be clever to reuse the `invokeValidators`since it also just need the `isOverThreeCharacters` rule. We end up with the following

```jsx
<label type="text" id="firstName" onChange={e => invokeValidators(e.target.value)} />
<label type="text" id="lastName" onChange={e => invokeValidators(e.target.value)} />
```

They would most likely run into the issue where the validation state is now shared between these two inputs when they intended to warn of invalid inputs separately.

Reply:
yes, I will add a few more usage examples.